src/transformers/models/imagebind/convert_imagebind_to_hf.py

```diff
@@ -0,0 +1,221 @@
+        ("modality_heads.vision.0.bias", "vision_model.layernorm.bias"),
+        ("modality_heads.vision.2.weight", "vision_projection.weight"),
+        ("modality_trunks.vision.pre_transformer_layer.0.weight", "vision_model.pre_layernorm.weight"),
+        ("modality_trunks.vision.pre_transformer_layer.0.bias", "vision_model.pre_layernorm.bias"),
+    ])
+
+    rename_keys.extend(
+        rename_encoder_layers(vision_config, "vision")
+    )
+
+    # Convert Text
+    rename_keys.extend([
+        ("modality_preprocessors.text.pos_embed", "text_model.embeddings.position_embedding.weight"),
+        ("modality_preprocessors.text.token_embedding.weight", "text_model.embeddings.token_embedding.weight"),
+        ("modality_heads.text.proj.0.weight", "text_model.layernorm.weight"),
+        ("modality_heads.text.proj.0.bias", "text_model.layernorm.bias"),
+        ("modality_heads.text.proj.1.weight", "text_projection.weight"),
+        ("modality_postprocessors.text.1.log_logit_scale", "text_postprocessor.log_logit_scale"),
+    ])
+
+    rename_keys.extend(
+        rename_encoder_layers(text_config, "text")
+    )
+
+    # Convert Audio
+    rename_keys.extend([
+        ("modality_preprocessors.audio.cls_token", "audio_model.embeddings.cls_token"),
+        ("modality_preprocessors.audio.rgbt_stem.proj.weight", "audio_model.embeddings.patch_embedding.projection.weight"),
+        ("modality_preprocessors.audio.rgbt_stem.norm_layer.weight", "audio_model.embeddings.patch_embedding.layernorm.weight"),
+        ("modality_preprocessors.audio.rgbt_stem.norm_layer.bias", "audio_model.embeddings.patch_embedding.layernorm.bias"),
+        ("modality_preprocessors.audio.pos_embedding_helper.pos_embed", "audio_model.embeddings.position_embeddings"),
+        ("modality_heads.audio.0.weight", "audio_model.layernorm.weight"),
+        ("modality_heads.audio.0.bias", "audio_model.layernorm.bias"),
+        ("modality_heads.audio.2.weight", "audio_projection.weight"),
+    ])
+
+    rename_keys.extend(
+        rename_encoder_layers(audio_config, "audio")
+    )
+    # fmt: on
+
+    return rename_keys
+
+
+def rename_key(dct, old, new):
+    val = dct.pop(old)
+    dct[new] = val
+
+
+def reshape_text_position_embeddings(state_dict):
+    # Need to convert from (1, contexc_length, hidden_size) -> (context_length, hidden_size)
+    position_embeddings = state_dict["text_model.embeddings.position_embedding.weight"]
+    state_dict["text_model.embeddings.position_embedding.weight"] = position_embeddings.squeeze(0)
+
+    return state_dict
+
+
+def prepare_input():
+    ds = load_dataset("EduardoPacheco/imagebind-example-data", split="train")
+    images = ds["image"]
+    texts = ds["text"]
+    audios = [
+        torchaudio.functional.resample(
+            torch.from_numpy(audio["array"]), orig_freq=audio["sampling_rate"], new_freq=16000
+        ).numpy()
+        for audio in ds["audio"]
+    ]
+
+    return images, texts, audios
```

Comment:
Looks like this is unused no ?

Reply:
Yes, it was used for converted model weights testing but when @molbap reviewed the files he suggested to move the assertion tests. They have been moved and this func can be omitted :+1: 

Status: Done in the recent commits